src/components/application_manager/rpc_plugins/vehicle_info_plugin/test/vehicle_info_plugin_test.cc

```diff
@@ -0,0 +1,238 @@
+  NiceMock<app_manager_test::MockApplicationManager> app_manager_mock_;
+  NiceMock<app_manager_test::MockRPCService> mock_rpc_service_;
+  NiceMock<app_manager_test::MockHMICapabilities> mock_hmi_capabilities_;
+  NiceMock<test::components::policy_test::MockPolicyHandlerInterface>
+      mock_policy_handler_;
+  MockMessageHelper& mock_message_helper_;
+  application_manager::ApplicationSet application_set_;
+  std::shared_ptr<sync_primitives::Lock> applications_lock_;
+  DataAccessor<application_manager::ApplicationSet> applications_;
+  std::shared_ptr<NiceMock<app_manager_test::MockApplication> > mock_app_ptr_;
+};
+
+TEST_F(VehicleInfoPluginTest, OnPolicyEvent_NoSubscriptions_Unsuccess) {
+  EXPECT_CALL(app_manager_mock_, applications())
+      .WillOnce(Return(applications_));
+  EXPECT_CALL(app_manager_mock_, GetRPCService()).Times(0);
+
+  vehicle_info_plugin_->OnPolicyEvent(
+      plugins::PolicyEvent::kApplicationsDisabled);
+}
+
+TEST_F(VehicleInfoPluginTest, OnPolicyEvent_SubscriptionExists_Unsuccess) {
+  application_manager::ApplicationSet application_set;
+  application_set.insert(mock_app_ptr_);
+  auto applications_lock = std::make_shared<sync_primitives::Lock>();
+  DataAccessor<application_manager::ApplicationSet> applications(
+      application_set, applications_lock);
+  EXPECT_CALL(app_manager_mock_, applications()).WillOnce(Return(applications));
+
+  VehicleInfoAppExtension app_extension(*vehicle_info_plugin_, *mock_app_ptr_);
+  auto extension_ptr = std::make_shared<VehicleInfoAppExtension>(app_extension);
+  ON_CALL(*mock_app_ptr_, QueryInterface(_))
+      .WillByDefault(Return(extension_ptr));
+
+  EXPECT_CALL(app_manager_mock_, GetRPCService()).Times(0);
+
+  vehicle_info_plugin_->OnPolicyEvent(
+      plugins::PolicyEvent::kApplicationsDisabled);
+}
+
+TEST_F(VehicleInfoPluginTest, OnApplicationEvent_AppRegistered_Success) {
+  EXPECT_CALL(*mock_app_ptr_, AddExtension(_)).WillOnce(Return(true));
+  vehicle_info_plugin_->OnApplicationEvent(
+      plugins::ApplicationEvent::kApplicationRegistered, mock_app_ptr_);
+}
+
+TEST_F(VehicleInfoPluginTest,
+       OnApplicationEvent_AppUnregisteredNoSubscriptions_Unsuccess) {
+  VehicleInfoAppExtension app_extension(*vehicle_info_plugin_, *mock_app_ptr_);
+  auto extension_ptr = std::make_shared<VehicleInfoAppExtension>(app_extension);
+  ON_CALL(*mock_app_ptr_, QueryInterface(_))
+      .WillByDefault(Return(extension_ptr));
+
+  EXPECT_CALL(app_manager_mock_, GetRPCService()).Times(0);
+  vehicle_info_plugin_->OnApplicationEvent(
+      plugins::ApplicationEvent::kApplicationUnregistered, mock_app_ptr_);
+}
+
+TEST_F(VehicleInfoPluginTest,
+       OnApplicationEvent_DeleteAppDataSendCommandsToHmi_Success) {
+  VehicleInfoAppExtension app_extension(*vehicle_info_plugin_, *mock_app_ptr_);
+  app_extension.subscribeToVehicleInfo(kVehicleData);
+  auto extension_ptr = std::make_shared<VehicleInfoAppExtension>(app_extension);
+  ON_CALL(*mock_app_ptr_, QueryInterface(_))
```

Comment:
Why parameter of QueryInterface is not specified?

Reply:
@YarikMamykin Because it cannot change and already defined in code, that's why I skip it for short. But I can check it, if needed. What's your opinion of this?

Reply:
@ychernysheva I think it should be better to verify particular parameter that is passed through `QueryInterface()` method, so please add it.

Reply:
@YarikMamykin Done in 995e90e0aa250402de566d3e4af7e4f04691ec25